Frontend/src/app/query-interface/properties/page.tsx

```diff
@@ -0,0 +1,181 @@
+import {
+  theme,
+  Button,
+  Form,
+  Input,
+  InputNumber,
+  Select,
+  Switch,
+  message,
+} from 'antd';
+import Box from '@mui/material/Box';
+import LinearProgress from '@mui/material/LinearProgress';
+import { Analyzers, GraphAnalyzer, InputParam, InputTypes } from "@/data/analyze-data";
+import { getGraphList } from "@/services/graph-service";
+import { ISelectProp } from "@/types/user-types";
+import type { FormProps } from 'antd';
+import { analyzeGraph } from "@/services/analyzer-service";
+
+// Render input field based on InputTypes
+const renderInputField = (key: string, inputParam: InputParam) => {
+  switch (inputParam.type) {
+    case InputTypes.INPUT:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <Input />
+        </Form.Item>
+      );
+    case InputTypes.INPUTNUMBER:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <InputNumber />
+        </Form.Item>
+      );
+    case InputTypes.SWITCH:
+      return (
+        <Form.Item label={inputParam.label || key} name={key} valuePropName="checked">
+          <Switch />
+        </Form.Item>
+      );
+    case InputTypes.SELECT:
+      return (
+        <Form.Item label={inputParam.label || key} name={key}>
+          <Select>
+            {inputParam.options?.map((option, index) => (
+              <Select.Option key={index} value={option.value}>
+                {option.label}
+              </Select.Option>
+            ))}
+          </Select>
+        </Form.Item>
+      );
+    default:
+      return null;
+  }
+};
+
+export default function QueryPropoerties() {
+  const { token } = theme.useToken();
+
+  const [graphs, setGraphs] = useState<ISelectProp[]>([]);
+  const [isAnalysing, setIsAnalysing] = useState<boolean>(false);
+  const [selectedTool, setSelectedTool] = useState<GraphAnalyzer>();
+
+  const getGraphsData = async () => {
+    try{
+    const res = await getGraphList();
+    console.log("::res::", res)
+    if(res.data){
+      const filteredData: ISelectProp[] = res.data.map((graph: any) => {
+         return {
+          value: graph.idgraph,
+          label: graph.name
+        }
+      })
+      setGraphs(filteredData);
+      console.log(filteredData)
+    }
+    }catch(err){
+      message.error("Failed to fetch graphs");
+    }
+  }
+
+  useEffect(() => {
+    getGraphsData();
+  }, [])
+
+  const listStyle: React.CSSProperties = {
+    lineHeight: '200px',
+    textAlign: 'center',
+    background: token.colorFillAlter,
+    borderRadius: token.borderRadiusLG,
+    marginTop: 16,
+  };
+  
+  type FieldType = {
+    graph_id?: string;
+    method?: string;
+  };
+
+  const runAnalysis = async (values: any) => {
+    console.log("::values::", values)
+    const result = await analyzeGraph(values);
+    if (result.data) {
+      message.success("Analysis completed successfully");
+    } else {
+      message.error(result.message);
```

Comment:
Is it sufficient just throwing the error message? Please double check this. I am wondering what kind of error message the user will see. May be better have message prefix "Error in graph analysis: " and append this `result.message` part to it.

Reply:
changed